src/pk/ecc/ecc_dp_find_by_name.c

```diff
@@ -0,0 +1,30 @@
+/* LibTomCrypt, modular cryptographic library -- Tom St Denis
+ *
+ * LibTomCrypt is a library that provides various cryptographic
+ * algorithms in a highly modular and flexible manner.
+ *
+ * The library is free for all purposes without any express
+ * guarantee it works.
+ */
+
+#include "tomcrypt.h"
+
+#ifdef LTC_MECC
+
+ltc_ecc_set_type* ecc_dp_find_by_name(char *curve_name)
+{
+   int i;
+
+   for (i = 0; ltc_ecc_sets[i].size != 0; i++) {
+      if (ltc_ecc_sets[i].name != NULL && XSTRCMP(ltc_ecc_sets[i].name, curve_name) == 0) {
+         return (ltc_ecc_set_type*)&ltc_ecc_sets[i];
```

Comment:
I have created some draft related to "possibility 2" 
* https://github.com/libtom/libtomcrypt/blob/pr/ecc-non-asn1-part/src/pk/ecc/ecc_dp_utils1.c

But it turned into a question: Why do we make copies of hexadecimal strings? It would make IMO more sense to convert those hex string into bignums during making a "copy":
* https://github.com/libtom/libtomcrypt/blob/pr/ecc-non-asn1-part/src/pk/ecc/ecc_dp_utils2.c